lib/pages/NewsPage.dart

```diff
@@ -0,0 +1,104 @@
+}
+
+class NewsPage extends StatefulWidget {
+  @override
+  _NewsPageState createState() => _NewsPageState();
+}
+
+class _NewsPageState extends State<NewsPage> {
+  final List<NewsMessage> newsMessages = [];
+
+  TextEditingController _messageController = TextEditingController();
+
+  void _showMessageDialog(BuildContext context) {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text('Zadejte zprávu'),
+          content: TextField(
+            controller: _messageController,
+            decoration: InputDecoration(hintText: 'Zpráva'),
+          ),
+          actions: [
+            TextButton(
+              onPressed: () => Navigator.pop(context),
+              child: Text('Storno'),
+            ),
+            TextButton(
+              onPressed: () {
+                String message = _messageController.text;
+                _sendMessage(message);
+                Navigator.pop(context);
+              },
+              child: Text('Odeslat'),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  void _sendMessage(String message) {
+    DateTime currentDate = DateTime.now();
+    NewsMessage newMessage = NewsMessage(date: currentDate, message: message);
+    setState(() {
+      newsMessages.insert(0, newMessage);
+    });
+    _messageController.clear();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Stránka s Ohláškami'),
+      ),
+      body: ListView.builder(
+        itemCount: newsMessages.length,
+        itemBuilder: (BuildContext context, int index) {
+          final message = newsMessages[index];
+          final previousMessage = index > 0 ? newsMessages[index - 1] : null;
+
+          final isSameDay = previousMessage != null &&
+              message.date.year == previousMessage.date.year &&
+              message.date.month == previousMessage.date.month &&
+              message.date.day == previousMessage.date.day;
+
+          return Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              if (index == 0 || !isSameDay)
+                Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 8.0, horizontal: 16.0),
+                  child: Text(
+                    formatDate(message.date),
+                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
+                  ),
+                ),
+              ListTile(
+                title: Text(message.message),
+              ),
+              Divider(),
+            ],
+          );
+        },
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () => _showMessageDialog(context),
+        child: Icon(Icons.add),
+      ),
+    );
+  }
+}
+
+String formatDate(DateTime date) {
+  return "${date.day}.${date.month}.${date.year}";
```

Comment:
could you add week day name in braces before the date? (In czech) e.g. "(čtvrtek) 20.5.2023"?